File: dqops/src/main/java/com/dqops/execution/statistics/jobs/CollectStatisticsResult.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.execution.statistics.jobs;

import com.dqops.execution.statistics.StatisticsCollectionExecutionSummary;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Returns the result with the summary of the statistics collected.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@ApiModel(value = "CollectStatisticsResult", description = "Returns the result with the summary of the statistics collected.")
@EqualsAndHashCode(callSuper = false)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@Data
public class CollectStatisticsResult {
    // TODO: How are these two counts different?
    /**
     * The total count of all executed statistics collectors.
     */
    @JsonPropertyDescription("The total count of all executed statistics collectors.")
    private int executedStatisticsCollectors;

    /**
     * The count of columns for which DQOps executed a collector and tried to read the statistics.
     */
    @JsonPropertyDescription("The count of columns for which DQOps executed a collector and tried to read the statistics.")
    private int columnsAnalyzed;

    /**
     * The count of columns for which DQOps managed to obtain statistics.
     */
    @JsonPropertyDescription("The count of columns for which DQOps managed to obtain statistics.")
    private int columnsSuccessfullyAnalyzed;

    /**
     * The count of statistics collectors that failed to execute.
     */
    @JsonPropertyDescription("The count of statistics collectors that failed to execute.")
    private int totalCollectorsFailed;

    /**
     * The total number of results that were collected.
     */
    @JsonPropertyDescription("The total number of results that were collected.")
    private int totalCollectedResults;

    /**
     * The default parameterless constructor.
     */
    public CollectStatisticsResult() {
    }

    /**
     * Creates a collect statistics job result from the execution summary.
     * @param statisticsCollectionExecutionSummary Statistics collection execution summary.
     * @return The job result object.
     */
    public static CollectStatisticsResult fromStatisticsExecutionSummary(StatisticsCollectionExecutionSummary statisticsCollectionExecutionSummary) {
        CollectStatisticsResult runChecksQueueJobResult = new CollectStatisticsResult() {{
            setExecutedStatisticsCollectors(statisticsCollectionExecutionSummary.getTotalCollectorsExecuted());
            setColumnsAnalyzed(statisticsCollectionExecutionSummary.getColumnsAnalyzedCount());
            setColumnsSuccessfullyAnalyzed(statisticsCollectionExecutionSummary.getColumnsSuccessfullyAnalyzed());
            setTotalCollectorsFailed(statisticsCollectionExecutionSummary.getTotalCollectorsFailed());
            setTotalCollectedResults(statisticsCollectionExecutionSummary.getTotalCollectedResults());
        }};

        return runChecksQueueJobResult;
    }

    public static class CollectStatisticsResultSampleFactory implements SampleValueFactory<CollectStatisticsResult> {
        @Override
        public CollectStatisticsResult createSample() {
            // Columns analyzed, 1*3 + 2*2 + 4*1
            // One of the sensors failed for a 3 statistics column.
            // One for one of the two 2-statistics columns.
            // One for one of the three 1-statistic columns.
            return new CollectStatisticsResult() {{
                setExecutedStatisticsCollectors(11);
                setColumnsAnalyzed(7);
                setColumnsSuccessfullyAnalyzed(4);
                setTotalCollectorsFailed(3);
                setTotalCollectedResults(8);
            }};
        }
    }
}

File: distribution/python/dqops/client/models/collect_statistics_result.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="CollectStatisticsResult")


@_attrs_define
class CollectStatisticsResult:
    """Returns the result with the summary of the statistics collected.

    Attributes:
        executed_statistics_collectors (Union[Unset, int]): The total count of all executed statistics collectors.
        columns_analyzed (Union[Unset, int]): The count of columns for which DQOps executed a collector and tried to
            read the statistics.
        columns_successfully_analyzed (Union[Unset, int]): The count of columns for which DQOps managed to obtain
            statistics.
        total_collectors_failed (Union[Unset, int]): The count of statistics collectors that failed to execute.
        total_collected_results (Union[Unset, int]): The total number of results that were collected.
    """

    executed_statistics_collectors: Union[Unset, int] = UNSET
    columns_analyzed: Union[Unset, int] = UNSET
    columns_successfully_analyzed: Union[Unset, int] = UNSET
    total_collectors_failed: Union[Unset, int] = UNSET
    total_collected_results: Union[Unset, int] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        executed_statistics_collectors = self.executed_statistics_collectors
        columns_analyzed = self.columns_analyzed
        columns_successfully_analyzed = self.columns_successfully_analyzed
        total_collectors_failed = self.total_collectors_failed
        total_collected_results = self.total_collected_results

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if executed_statistics_collectors is not UNSET:
            field_dict["executed_statistics_collectors"] = (
                executed_statistics_collectors
            )
        if columns_analyzed is not UNSET:
            field_dict["columns_analyzed"] = columns_analyzed
        if columns_successfully_analyzed is not UNSET:
            field_dict["columns_successfully_analyzed"] = columns_successfully_analyzed
        if total_collectors_failed is not UNSET:
            field_dict["total_collectors_failed"] = total_collectors_failed
        if total_collected_results is not UNSET:
            field_dict["total_collected_results"] = total_collected_results

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        executed_statistics_collectors = d.pop("executed_statistics_collectors", UNSET)

        columns_analyzed = d.pop("columns_analyzed", UNSET)

        columns_successfully_analyzed = d.pop("columns_successfully_analyzed", UNSET)

        total_collectors_failed = d.pop("total_collectors_failed", UNSET)

        total_collected_results = d.pop("total_collected_results", UNSET)

        collect_statistics_result = cls(
            executed_statistics_collectors=executed_statistics_collectors,
            columns_analyzed=columns_analyzed,
            columns_successfully_analyzed=columns_successfully_analyzed,
            total_collectors_failed=total_collectors_failed,
            total_collected_results=total_collected_results,
        )

        collect_statistics_result.additional_properties = d
        return collect_statistics_result

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties
